StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Components/Toast/LinkToast.swift

```diff
@@ -0,0 +1,187 @@
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setup() {
+        backgroundColor = .linkToastBackground
+        directionalLayoutMargins = .insets(amount: Constants.padding)
+
+        insetsLayoutMarginsFromSafeArea = false
+
+        label.text = text
+        iconView.image = toastType.icon
+        iconView.tintColor = toastType.iconColor
+
+        let stackView = UIStackView(arrangedSubviews: [iconView, label])
+        stackView.spacing = Constants.spacing
+        stackView.alignment = .center
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+
+        addSubview(stackView)
+
+        NSLayoutConstraint.activate([
+            stackView.topAnchor.constraint(equalTo: layoutMarginsGuide.topAnchor),
+            stackView.bottomAnchor.constraint(equalTo: layoutMarginsGuide.bottomAnchor),
+            stackView.leadingAnchor.constraint(equalTo: layoutMarginsGuide.leadingAnchor),
+            stackView.trailingAnchor.constraint(equalTo: layoutMarginsGuide.trailingAnchor),
+        ])
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        layer.cornerRadius = bounds.height / 2
+    }
+
+}
+
+// MARK: - Show/Hide
+
+extension LinkToast {
+
+    /// Show the toast from the given view as context.
+    /// - Parameters:
+    ///   - view: View to show the toast from.
+    ///   - duration: How long to show the toast for.
+    func show(from view: UIView, duration: TimeInterval = Constants.defaultDuration) {
+        let presentingView = view.window ?? view
+
+        translatesAutoresizingMaskIntoConstraints = false
+        presentingView.addSubview(self)
+
+        NSLayoutConstraint.activate([
+            // Horizontally center
+            centerXAnchor.constraint(equalTo: presentingView.safeAreaLayoutGuide.centerXAnchor),
+
+            // Pin edges
+            topAnchor.constraint(equalTo: presentingView.safeAreaLayoutGuide.topAnchor),
+            leadingAnchor.constraint(greaterThanOrEqualTo: presentingView.layoutMarginsGuide.leadingAnchor),
+            trailingAnchor.constraint(lessThanOrEqualTo: presentingView.layoutMarginsGuide.trailingAnchor),
+        ])
+
+        alpha = 0
+        transform = CGAffineTransform(translationX: 0, y: -Constants.animationTravelDistance)
+
+        UIView.animate(
+            withDuration: Constants.animationDuration,
+            delay: 0,
+            options: .curveEaseOut
+        ) {
+            self.alpha = 1
+            self.transform = .identity
+        }
+
+        UIAccessibility.post(notification: .announcement, argument: text)
+        #if !os(visionOS)
+        generateHapticFeedback()
+        #endif
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + duration) {
+            self.hide()
+        }
+    }
+
+    /// Hides toast.
+    ///
+    /// You normally don't need to call this, as the toast will hide on its own after an specific duration.
```

Comment:
This comment is vague-- it explains that we don't normally need to call this, but doesn't describe when we should be calling it.

Reply:
I think it's for if you want to hide it manually. Like maybe you have a "success" toast, then an error immediately occurs and you want to cancel the first toast?